packages/compass-web/sandbox/index.tsx

```diff
@@ -241,46 +242,65 @@ function ConnectionStringConnectionForm({
 
 function ConnectedApp({ connectionInfo }: { connectionInfo: ConnectionInfo }) {
   const isAtlasConnection = !!connectionInfo.atlasMetadata;
-  const [initialCurrentTab, updateCurrentTab] = useWorkspaceTabRouter(
-    connectionInfo?.id
+  const [currentTab, updateCurrentTab] = useWorkspaceTabRouter(
+    connectionInfo.id
+  );
+  const initialCurrentTabRef = useRef(
+    currentTab ?? {
+      type: 'Databases' as const,
+      connectionId: connectionInfo.id,
+    }
   );
 
+  const atlasServiceSandboxBackendVariant =
+    process.env.COMPASS_WEB_HTTP_PROXY_CLOUD_CONFIG === 'local'
+      ? 'web-sandbox-atlas-local'
+      : process.env.COMPASS_WEB_HTTP_PROXY_CLOUD_CONFIG === 'dev' ||
+        process.env.COMPASS_WEB_HTTP_PROXY_CLOUD_CONFIG === 'qa'
+      ? 'web-sandbox-atlas-dev'
+      : 'web-sandbox-atlas';
+
   return (
-    <Body as="div" className={sandboxContainerStyles}>
-      <LoggerAndTelemetryProvider value={sandboxLogger}>
-        <CompassWeb
-          onAutoconnectInfoRequest={() => {
-            return Promise.resolve(connectionInfo);
-          }}
-          initialWorkspaceTabs={
-            initialCurrentTab ? [initialCurrentTab] : undefined
-          }
-          onActiveWorkspaceTabChange={updateCurrentTab}
-          initialPreferences={{
-            enablePerformanceAdvisorBanner: isAtlasConnection,
-            enableAtlasSearchIndexes: !isAtlasConnection,
-            maximumNumberOfActiveConnections: isAtlasConnection ? 1 : 10,
-          }}
-          stackedElementsZIndex={5}
-          renderConnecting={(connectionInfo) => {
-            return (
-              <LoadingScreen
-                connectionString={
-                  connectionInfo?.connectionOptions.connectionString
-                }
-              ></LoadingScreen>
-            );
-          }}
-          renderError={(_connectionInfo, err) => {
-            return (
-              <ErrorScreen
-                error={err.message ?? 'Error occured when connecting'}
-              ></ErrorScreen>
-            );
-          }}
-        ></CompassWeb>
-      </LoggerAndTelemetryProvider>
-    </Body>
+    <SandboxAutoconnectProvider
+      // Even for the sanbos we only pass the connection info here when not
+      // connecting to atlas cluster as atlas connection should resolve the
+      // connection info automatically through the new connection storage
+      value={!isAtlasConnection ? connectionInfo : null}
+    >
+      <Body as="div" className={sandboxContainerStyles}>
+        <LoggerAndTelemetryProvider value={sandboxLogger}>
+          <CompassWeb
+            orgId={connectionInfo.atlasMetadata?.orgId ?? ''}
```

Comment:
[q] Are we able to differentiate between when there's no atlasMetadata because it's not connecting to an atlas cluster versus when there _should_ be atlasMetadata in the connectionInfo but it's not passed through? Just wondering

Reply:
Good question! In sandbox we can always be sure that if we get the `atlasMetadata` provided on the `connectionInfo`, we are connecting to atlas cluster, it is a bit hard to communicate in the code though, yeah...

Reply:
Rewrote this code a bit in https://github.com/mongodb-js/compass/pull/5863/commits/c256b6562228ca9174befefd1e495826a2eab314 trying to make it a bit clearer in the code what exactly is happening here